hooks/pre-commit

```diff
@@ -13,3 +13,22 @@ then
   echo Please commit to a working branch and push it to GitHub for review/merge.
   exit 1
 fi
+
+readonly changed_files_command="git diff --cached --name-only | grep '\.py$' | grep -v 'servers/[^\/]*/jobs/models' | grep -v 'setup.py'"
+readonly files_in_commit=$(eval $changed_files_command)
+if [ -n "$files_in_commit" ]
+then
+  # Stash un-staged changes to ensure that the staged changes pass the linter
+  git stash -q --keep-index
+  readonly yapf_out=$(yapf -d $files_in_commit)
+
+  if ! [ -z "$yapf_out" ]
+  then
+    echo "Reformatting staged files, please review them and stage the changes."
```

Comment:
I just tested this out.
- Made a change to a file that would cause yapf to fail
- git commit -m 'blah'
- commit went through with incorrectly formatted file
- After commit, file is still incorrectly formatted.

That's not what you expect, right?

My advice is to abandon this PR, rather than try to get this working.

If you can get the automatic reformatting working, then this PR is worth it. Seems doable given you have file_in_commit.